### ISSUE TITLE
How to set correct Distances Source To Detector and Source To Patient

### ISSUE THREAD

**Issue by equamo** [eigenvivek/DiffDRR#372]
Hi together and hi @eigenvivek , 

I want to make a comparison of the DRR's to real generated x-rays.
From the original x-rays I know the Distances Source To Detector and Source To Patient. For example, in one case the parameters are:
DistanceSourceToDetector=1800.0, DistanceSourceToPatient=1746.0

How do I have to set these parameters in code?
I guess for source to detector its obvious, we have to set the sdd parameter in the DRR constructor. But how can I achieve that DistanceSourceToPatient is correct as well? Can you tell me the correct parameters for these sample distances?

Best

**Comment by eigenvivek** [comment]
hi @equamo here's an example of how to put SDD and DSO into the DRR renderer

```python
import matplotlib.pyplot as plt
import torch

from diffdrr.drr import DRR
from diffdrr.data import load_example_ct
from diffdrr.visualization import plot_drr

DistanceSourceToDetector = 1800.0
DistanceSourceToPatient = 1746.0

# Read in the volume and get its origin and spacing in world coordinates
subject = load_example_ct()

# Initialize the DRR module for generating synthetic X-rays
device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
drr = DRR(
    subject,     # An object storing the CT volume, origin, and voxel spacing
    sdd=DistanceSourceToDetector,  # Source-to-detector distance (i.e., focal length)
    height=200,  # Image height (if width is not provided, the generated DRR is square)
    delx=2.0,    # Pixel spacing (in mm)
).to(device)

# Set the camera pose with rotations (yaw, pitch, roll) and translations (x, y, z)
rotations = torch.tensor([[0.0, 0.0, 0.0]], device=device)
translations = torch.tensor([[0.0, DistanceSourceToPatient, 0.0]], device=device)

# 📸 Also note that DiffDRR can take many representations of SO(3) 📸
# For example, quaternions, rotation matrix, axis-angle, etc...
img = drr(rotations, translations, parameterization="euler_angles", convention="ZXY")
plot_drr(img, ticks=False)
plt.show()
```

**Comment by eigenvivek** [comment]
closing for now, feel free to open a new post if you have any more questions

**Comment by equamo** [comment]
Thank you, helped me a lot

**Comment by HumamHelfawi** [comment]
@eigenvivek  what about SAD (Source to axis distance). I did not find that DiffDRR using it at all. How is it possible to get geometrically accurate results without incorporating it?

**Comment by eigenvivek** [comment]
what’s the definition of source to axis distance?

**Comment by eigenvivek** [comment]
so `SDD = SID + SAD`?

then you'll want something like this: 

```python
rot = torch.tensor([[0.0, 0.0, 0.0]], device=device)
xyz = torch.tensor([[0.0, SAD, 0.0]], device=device)
img = drr(rot, xyz, parameterization="euler_angles", convention="ZXY")
```

**Comment by HumamHelfawi** [comment]
Actually SDD = SID and SAD seems like what you have used.

Thanks. It makes sense. that what I tired first but it gave me unexpected result comparing to another projection method I am working on. So I suspected using SAD like that was incorrect. Maybe the error is in other part of my projection code.
Thanks again for your help!

**Comment by eigenvivek** [comment]
no problem. feel free to open a new issue if you're having any issues in the future
